server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -196,6 +199,76 @@ public RestoreService(
         this.shardLimitValidator = shardLimitValidator;
     }
 
+    /**
+     * Restores data from remote store for indices specified in the restore request.
+     *
+     * @param request  restore request
+     * @param listener restore listener
+     */
+    public void restoreFromRemoteStore(RestoreRemoteStoreRequest request, final ActionListener<RestoreCompletionResponse> listener) {
```

Comment:
The method exists in the snapshot package, we should extend it and abstract it out

Reply:
Yes,`restoreSnapshot` method is in the same class. I earlier thought of extending it but `RestoreSnapshotRequest` is different than `RestoreRemoteStoreRequest` and even if we create one method, it will anyway have conditional check to redirect the flow to snapshot restore vs remote store restore.